security-analysis/src/main/java/com/powsybl/security/SecurityAnalysis.java

```diff
@@ -24,7 +25,17 @@ default boolean removeInterceptor(SecurityAnalysisInterceptor interceptor) {
         return false;
     }
 
-    CompletableFuture<SecurityAnalysisResult> runAsync(ContingenciesProvider contingenciesProvider, String workingStateId, LoadFlowParameters parameters);
+    /**
+     * @deprecated Use CompletableFuture<SecurityAnalysisResult> runAsync(ContingenciesProvider, String, LoadFlowParameters, SecurityAnalysisParameters) instead
+     */
+    @Deprecated
+    default CompletableFuture<SecurityAnalysisResult> runAsync(ContingenciesProvider contingenciesProvider, String workingStateId, LoadFlowParameters parameters) {
+        throw new UnsupportedOperationException("Unimplemented method.");
+    }
+
+    default CompletableFuture<SecurityAnalysisResult> runAsync(ContingenciesProvider contingenciesProvider, String workingStateId, LoadFlowParameters loadFlowParameters, SecurityAnalysisParameters securityAnalysisParameters) {
+        return runAsync(contingenciesProvider, workingStateId, loadFlowParameters);
+    }
```

Comment:
You don't have to create a default implementation for these existing methods. You have to give a default implementation for the new method.